usecases/evaluate_scenario/evaluate_scenario.go

```diff
@@ -252,13 +253,19 @@ func EvalScenario(
 	)
 	defer span.End()
 
-	liveVersion, err := repositories.EvalScenarioRepository.GetScenarioIteration(ctx, exec, *params.Scenario.LiveVersionID)
+	var targetVersionId string
+	if params.ScenarioIterationId != nil {
+		targetVersionId = *params.ScenarioIterationId
+	} else {
+		targetVersionId = *params.Scenario.LiveVersionID
+	}
```

Comment:
I would move this up and check that `targetVersionId` is not nil on 240 (+ call it `versionToRunId` based on the above naming)

Reply:
yep that makes sense 👍